src/workflow-engine/charms/temporal-worker/README.md

```diff
@@ -4,138 +4,184 @@ This folder provides the Temporal Workflows and Activities for the existing
+    [tutorial](<https://juju.is/docs/juju/tutorial#heading--set-up-create-your-test-environment>),
+       - **NOTE:** if you are deploying the charm with Terraform from outside
+        the cluster, you might need to configure the Juju controller as follows:
+
+         ```bash
+         juju bootstrap my-microk8s-cloud <controller-name> \
+             --config controller-service-type=loadbalancer \
+             --config controller-external-ips='[<nodeIP>,...]'
+         ```
+
+         <a name="external-ip"></a>
+
+         This is because
+         Terraform will rely on the controller's `api-endpoints` which, by
+         default, are only set with a `ClusterIP` and are thus
+         **not reachable** from outside the cluster. By using these
+         `--config` options, you can tell Kubernetes to expose the
+         controller on the VM's IP, which should
+         be reachable from your machine. References:
          - <https://bugs.launchpad.net/juju/+bug/1883006>
          - <https://juju.is/docs/juju/juju-bootstrap>
          - <https://discourse.charmhub.io/t/juju-add-k8s-in-openstack-no-route-to-controller/3625>
 
-## Deploying the charm with Terraform
+       or
+    2. use [our production one](../../README.md). In this case, you can simply
+    SSH into the OpenStack environment and run `juju clouds`, to get something
+    like:
+
+        ```
+        Clouds available on the client:
+        Cloud      Regions  Default    Type  Credentials  Source    Description
+        localhost  1        localhost  lxd   0            built-in  LXD Container Hypervisor
+        microk8s   0                   k8s   0            built-in  A local Kubernetes context
+        ```
 
-This is the preferred option.
+        - the Juju controller for this microk8s cloud will be on a dedicated
+        VM, so if you're an admin, just `ssh ubuntu@<rocks-ps6-juju-client-ip>`
+        (see how this instance is configured [here](../../README.md)). If you're
+        not an admin, read "[From your workstation](#from-your-workstation)" to
+        know how to register the existing controller in your workstation.
 
-### Additional prerequisites
+## Deploying the charm with Terraform
 
-- `terraform` (>=v1.5.5)
-- this TF configuration is designed to use the MicroK8s cluster deployed
-[here](../../README.md). Before proceeding, make sure you:
-  - connect to the VPN,
-  - forward all the necessary remote endpoints to your `localhost``, with:
+This is the preferred option (Terraform should already be installed, but if not,
+run `sudo snap install terraform --classic`).
 
-      ```bash
-      ssh -L 5000:<keystoneUrl>:<keystonePort> -L ...<username>@<internalOSServer>
-      ```
+### Additional prerequisites
 
-    this is needed because of Bastion,
-- (optional) if you are managing an already existing deployment, **you need**
+- **if** you are managing an already existing deployment, **you need**
```

Comment:
maybe add the file inside the .gitignore to avoid this no?

Reply:
it is already